src/components/download-button/download-button.js

```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import Button from '@mapbox/mr-ui/button';
+import IconText from '@mapbox/mr-ui/icon-text';
+import PropTypes from 'prop-types';
+
+export default class DownloadButton extends React.PureComponent {
+  render() {
+    const { fileType, href } = this.props;
+    const fileName = href
+      .toString()
+      .split('/')
+      .pop()
+      .replace(/\.\w+$/, '');
+
+    if (fileType === 'CSV') {
+      return (
+        <Button href={href}>
+          <IconText iconBefore="arrow-down">Download {fileType} file</IconText>
+        </Button>
+      );
+    } else {
+      return (
+        <Button href={href} passthroughProps={{ download: fileName }}>
+          <IconText iconBefore="arrow-down">Download {fileType} file</IconText>
+        </Button>
+      );
```

Comment:
There is repeated code here that I think we can optimize (repeated code often leads to human error and more maintenance). 

I see the difference between the two is that one has `passthroughProps`. 
- What might this look like if we added the React prop (`passthroughProps`) conditionally?
- Or, is there a downside to always setting `passthroughProps`? I don't actually know this, but if all file formats work with `passthroughProps`, then this would save us from having to add/maintain logic.

Reply:
>What might this look like if we added the React prop (passthroughProps) conditionally?

Good call! I reworked this to keep the code from repeating.
 
> Or, is there a downside to always setting passthroughProps? I don't actually know this, but if all file formats work with passthroughProps, then this would save us from having to add/maintain logic.

If I use `passthroughProps` with a CSV, the file extension is removed from the downloaded file. 

CSV with `passthroughProps` 👎 
<img width="740" alt="image" src="https://user-images.githubusercontent.com/19801577/130310493-e1d4e4c7-910c-4c7b-8a31-af860250c858.png">

CSV without `passthroughProps` 👍 
<img width="716" alt="image" src="https://user-images.githubusercontent.com/19801577/130310470-6dd4558d-a142-4b0a-8058-97301d905a92.png">

I can definitely do some more testing here!

Reply:
Good news! I think we can simplify this a bit more and remove the need for the file type conditional as you suggested. Using `passthroughProps` with the filename (including extension) for _all_ file types seems to be the best option here. 

If I remove `passthroughProps` entirely, the PDF and PNG will open in a new browser window rather than downloading directly. I think the above CSV issue was happening because the browser wasn't detecting and appending the file type with just the filename specified in the `download` attribute (just for CSVs it seems). When using `passthroughProps` with the filename and extension, the extension is added as expected. This works for all test cases!

I guess the only thing to look into further would be if there are any downsides to `passthroughProps` 🕵️